pinning.go

```diff
@@ -98,12 +98,25 @@ func (s *Server) doPinning(ctx context.Context, op *pinner.PinningOperation, cb
 	ctx, span := s.tracer.Start(ctx, "doPinning")
 	defer span.End()
 
+	connectedToAtLeastOne := false
 	for _, pi := range op.Peers {
 		if err := s.Node.Host.Connect(ctx, pi); err != nil {
 			log.Warnf("failed to connect to origin node for pinning operation: %s", err)
+		} else {
+			//	Check if it's trying to connect to it's own since we only want to check if the
+			//	the connection is between the host and the external/other peers.
+			if s.Node.Host.ID() != pi.ID {
+				connectedToAtLeastOne = true
+			}
 		}
 	}
 
+	//	If it can't connect to any legitimate provider peers, then we fail the entire operation.
+	if !connectedToAtLeastOne {
+		log.Errorf("unable to connect to any of the provider peers for pinning operation")
+		return nil
+	}
+
```

Comment:
If all the addresses refer to itself we want the code to succeed. This makes it fail.

Reply:
We can just remove the condition that checks the host.ID() and the pi.ID. This means the process will only proceed if it's connected to any providers including itself.